bundle-3.0/pom.xml

```diff
@@ -0,0 +1,1329 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2017 ABSA Group Limited
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>spark-3.0-spline-agent-bundle_2.11</artifactId>
```

Comment:
`scala-cross-build:restore-version` always changes it to `2.11`. Is there some way to override the default Scala version just for this bundle?

Reply:
hm.. it shouldn't. This module is only connected to Maven when you activate `scala-2.12` profile. So whatever plugin you execute, unless `scala-2.12` profile is activated this POM should stay disconnected and hence unaffected.

Reply:
try to remove all `pom.xml.backup` and try again.

Reply:
`scala-cross-build:restore-version -P scala-2.12` sets the suffix to _2.11 in spark 3 bundle. No matter what was there on the start. 

`scala-cross-build:restore-version -P scala-2.11` doesn't change spark 3 bundle.

I don't think the backups are used at all. I think the plugin just takes the default version and overwrite the suffixies by it.

Reply:
Why would you call `restore` with a specific profile? To me the command only makes sense when is called alone. The case when a set of modules is different depending on which Scala version profile you choose doesn't seem to be covered by the plugin. So I'd say simply don't call `restore` and instead always use `change-version`.
The main point here is that the `pom.xml` should by default be in the valid state, i.e. spark-3.0_2.12`

Reply:
It would be nice to have a command that would restore everything to default version, so I can erase any changes due to version switching. (for example before committing changes) But that doesn't seem to be possible right now.